webapp/channels/src/components/onboarding_tasklist/onboarding_tasklist.tsx

```diff
@@ -137,7 +137,7 @@ const OnBoardingTaskList = (): JSX.Element | null => {
     }, []);
 
     const open = useSelector(((state: GlobalState) => getBool(state, OnboardingTaskCategory, OnboardingTaskList.ONBOARDING_TASK_LIST_OPEN)));
-    const trigger = useRef<HTMLButtonElement>(null);
+    const [trigger, setTrigger] = useState<HTMLButtonElement | null>(null);
```

Comment:
This change seems wrong. Why would you want to re-render the component when the reference is added?

Reply:
We want it to re-render because Floating UI needs to wait until the `Button` is rendered and on the DOM to calculate the floating element's position. Normally, we'd just pass the `setReference` returned by `useFloating` directly into the button's `ref` prop, but we need this extra step because the `Button` is located outside of that component. Another option to do that would've been to make a hook for `OnboardingTaskList` like `useEmojiPicker` because a `useOnboardingTaskList` could return `setReference` here, but that would've required a lot more refactoring.

I think Floating UI actually uses a `useState` internally to store the referenced element as well, so this sort of mirrors what they do